ResolvePackageFileConflicts throwing when referencing System.Diagnostics.DiagnosticSource by assembly path

This is new failure that just started happening in some Razor tests for our MSBuild functionality. This started failing for us `2.2.0-preview1-007927` - this was working before in `2.2.0-preview1-007866`.

I've attached a repro project, which is doing some esoteric things because this is part of our tests for our MSBuild targets 😆. I think the probably might not be anything wierd that we're doing though.

The project file looks like this:
```
<Project Sdk="Microsoft.NET.Sdk.Web">
  <PropertyGroup>
    <TargetFramework>netcoreapp2.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup Condition="'$(BinariesRoot)'==''">
    <!-- In test scenarios $(BinariesRoot) is defined in a generated Directory.Build.props file -->
    <ProjectReference Include="..\..\Microsoft.AspNetCore.Razor.Test.MvcShim\Microsoft.AspNetCore.Razor.Test.MvcShim.csproj"/>
    <ProjectReference Include="..\..\..\src\Microsoft.AspNetCore.Razor.Runtime\Microsoft.AspNetCore.Razor.Runtime.csproj"/>
  </ItemGroup>

  <ItemGroup Condition="'$(BinariesRoot)'!=''">
    <Reference Include="$(BinariesRoot)\System.Diagnostics.DiagnosticSource.dll"/>
    <Reference Include="$(BinariesRoot)\Microsoft.AspNetCore.Html.Abstractions.dll"/>
    <Reference Include="$(BinariesRoot)\Microsoft.AspNetCore.Razor.dll"/>
    <Reference Include="$(BinariesRoot)\Microsoft.AspNetCore.Razor.Runtime.dll"/>
    <Reference Include="$(BinariesRoot)\Microsoft.AspNetCore.Razor.Test.MvcShim.dll"/>
  </ItemGroup>
</Project>
```
Removing `<Reference Include="$(BinariesRoot)\System.Diagnostics.DiagnosticSource.dll"/>` seems to fix it.


```
Microsoft (R) Build Engine version 15.6.22.57775 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restoring packages for C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj...
  Generating MSBuild file C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\obj\SimpleMvc.csproj.nuget.g.props.
  Generating MSBuild file C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\obj\SimpleMvc.csproj.nuget.g.targets.
  Restore completed in 273.82 ms for C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj.
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018: The "ResolvePackageFileConflicts" task failed unexpectedly. [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018: System.ArgumentNullException: Value cannot be null. [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018: Parameter name: key [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at System.Collections.Generic.Dictionary`2.FindEntry(TKey key) [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at System.Collections.Generic.Dictionary`2.TryGetValue(TKey key, TValue& value) [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at Microsoft.NET.Build.Tasks.ConflictResolution.PackageOverrideResolver`1.Resolve(TConflictItem item1, TConflictItem item2) [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at Microsoft.NET.Build.Tasks.ConflictResolution.ConflictResolver`1.ResolveConflict(TConflictItem item1, TConflictItem item2) [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at Microsoft.NET.Build.Tasks.ConflictResolution.ConflictResolver`1.ResolveConflicts(IEnumerable`1 conflictItems, Func`2 getItemKey, Action`1 foundConflict, Boolean commitWinner, Action`1 unresolvedConflict) [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at Microsoft.NET.Build.Tasks.ConflictResolution.ResolvePackageFileConflicts.ExecuteCore()[C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at Microsoft.NET.Build.Tasks.TaskBase.Execute() [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() in E:\A\_work\24\s\src\Build\BackEnd\TaskExecutionHost\TaskExecutionHost.cs:line 631 [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.ConflictResolution.targets(59,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__25.MoveNext() in E:\A\_work\24\s\src\Build\BackEnd\Components\RequestBuilder\TaskBuilder.cs:line 787 [C:\Users\rynowak\AppData\Local\Temp\Razor\sbp0bspc.b03\SimpleMvc\SimpleMvc.csproj]
```


```
.NET Command Line Tools (2.2.0-preview1-007927)

Product Information:
 Version:            2.2.0-preview1-007927
 Commit SHA-1 hash:  a212d5a017

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Users\rynowak\.dotnet\x64\sdk\2.2.0-preview1-007927\

Microsoft .NET Core Shared Framework Host

  Version  : 2.1.0-preview1-26110-02
  Build    : 77152705cd56e7d89f52585959ec3559d91084c4
```
[Repro.zip](https://github.com/dotnet/sdk/files/1627368/Repro.zip)

--- a/src/Tasks/Common/ConflictResolution/PackageOverrideResolver.cs
+++ b/src/Tasks/Common/ConflictResolution/PackageOverrideResolver.cs
@@ -78,20 +78,18 @@ namespace Microsoft.NET.Build.Tasks.ConflictResolution
 
         public TConflictItem Resolve(TConflictItem item1, TConflictItem item2)
         {
-            if (PackageOverrides != null)
+            if (PackageOverrides != null && item1.PackageId != null && item2.PackageId != null)
             {
                 PackageOverride packageOverride;
                 Version version;
-                if (item1.PackageId != null
-                    && PackageOverrides.TryGetValue(item1.PackageId, out packageOverride)
+                if (PackageOverrides.TryGetValue(item1.PackageId, out packageOverride)
                     && packageOverride.OverriddenPackages.TryGetValue(item2.PackageId, out version)
                     && item2.PackageVersion != null
                     && item2.PackageVersion <= version)
                 {
                     return item1;
                 }
-                else if (item2.PackageId != null
-                    && PackageOverrides.TryGetValue(item2.PackageId, out packageOverride)
+                else if (PackageOverrides.TryGetValue(item2.PackageId, out packageOverride)
                     && packageOverride.OverriddenPackages.TryGetValue(item1.PackageId, out version)
                     && item1.PackageVersion != null
                     && item1.PackageVersion <= version)

--- a/src/Tasks/Common/NuGetUtils.cs
+++ b/src/Tasks/Common/NuGetUtils.cs
@@ -36,6 +36,13 @@ namespace Microsoft.NET.Build.Tasks
             {
                 // this method is just a temporary heuristic until we flow the NuGet metadata through the right items
                 // https://github.com/dotnet/sdk/issues/1091
+
+                // Don't try to recurse a relative path.
+                if (!Path.IsPathRooted(fullPath))
+                {
+                    return;
+                }
+
                 for (var dir = Directory.GetParent(fullPath); dir != null; dir = dir.Parent)
                 {
                     var nuspecs = dir.GetFiles("*.nuspec");

--- a/src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAPackageOverrideResolver.cs
+++ b/src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAPackageOverrideResolver.cs
@@ -48,5 +48,50 @@ namespace Microsoft.NET.Build.Tasks.UnitTests
             Assert.Null(resolver.PackageOverrides);
             Assert.Null(resolver.Resolve(new MockConflictItem(), new MockConflictItem()));
         }
+
+        [Fact]
+        public void ItHandlesNullPackageIds()
+        {
+            ITaskItem[] packageOverrides = new[]
+            {
+                new MockTaskItem("FakePlatform", new Dictionary<string, string>
+                {
+                    { MetadataKeys.OverriddenPackages, "System.Ben|4.2.0;System.Jobst-Immo|4.2.0;System.Livar|4.3.0;System.Dave|4.2.0" }
+                })
+            };
+
+            var resolver = new PackageOverrideResolver<MockConflictItem>(packageOverrides);
+
+            Assert.NotNull(resolver.PackageOverrides);
+
+            var packageItem = new MockConflictItem("System.Eric")
+            {
+                PackageId = "System.Eric",
+                PackageVersion = new Version(4, 0, 0),
+                AssemblyVersion = new Version(4, 0, 0, 0),
+                ItemType = ConflictItemType.Reference
+            };
+
+            var platformItem = new MockConflictItem("System.Eric")
+            {
+                PackageId = null,
+                AssemblyVersion = new Version(4, 1, 0, 0),
+                ItemType = ConflictItemType.Platform
+            };
+            
+            Assert.Null(resolver.Resolve(packageItem, platformItem));
+            Assert.Null(resolver.Resolve(platformItem, packageItem));
+
+            var packageItem2 = new MockConflictItem("System.Eric")
+            {
+                PackageId = "FakePlatform",
+                PackageVersion = new Version(4, 0, 0),
+                AssemblyVersion = new Version(4, 0, 0, 0),
+                ItemType = ConflictItemType.Reference
+            };
+
+            Assert.Null(resolver.Resolve(packageItem2, platformItem));
+            Assert.Null(resolver.Resolve(platformItem, packageItem2));
+        }
     }
 }

